common/web/keyboard-processor/src/keyboards/activeLayout.ts

```diff
@@ -539,11 +554,16 @@ export class ActiveRow implements LayoutRow {
   ) {
     // Apply defaults, setting the width and other undefined properties for each key
     let keys=row['key'];
+    const DEFAULT_KEY = ActiveKeyBase.DEFAULT_KEY;
     for(let j=0; j<keys.length; j++) {
       let key=keys[j];
-      for(var tp in ActiveKey.DEFAULT_KEY) {
-        if(typeof key[tp] != 'string' && typeof key[tp] != 'number') {
-          key[tp]=ActiveKey.DEFAULT_KEY[tp];
+      let keySet = Object.keys(DEFAULT_KEY);
+      for(let tp in keySet) {
+        const typedKey = tp as keyof typeof DEFAULT_KEY;
+        if(typeof key[typedKey] != 'string' && typeof key[typedKey] != 'number') {
+          // We detected a value of the wrong type.
+          // @ts-ignore  // Type 'string' is not assignable to type 'never'. (ts2322)
```

Comment:
This extra case of `@ts-ignore` is due to different reasons, and I felt the need to make that clear while extracting the changes related to the refactor.  So, it got "pulled up" into this PR.